status/condition.go

```diff
@@ -0,0 +1,58 @@
+// Inspired by https://github.com/knative/pkg/tree/97c7258e3a98b81459936bc7a29dc6a9540fa357/apis,
+// but we chose to diverge due to the unacceptably large dependency closure of knative/pkg.
+package status
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Object interface {
+	client.Object
+	GetConditions() []Condition
```

Comment:
I much prefer the explicit interface. It allows us to do things like manage pod conditions, as well, since we can adapt the conditiosn in these setters/getters.

Reply:
Doesn't pod store the status conditions in the same place as other objects like `status.conditions`? Not sure that I get your point but I'm also fine with evolving this over time

Reply:
It's not the same struct. It's similar, but predates the standard struct: https://pkg.go.dev/k8s.io/api/core/v1#PodCondition